rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1828,6 +1828,451 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
+		else
+			empty
+		
+func Create_ChangeRepurchasePriceInstruction: <"This function is used when the cash amount and collateral amount need to be modified. Thus all parameters are required.">
+	inputs:
+		tradeState TradeState (1..1) <"The original trade state and trade to be repriced.">
+		newAllinPrice number(1..1) <"The collateral new all-in price.">
+		newCashValue number(1..1)<"The new cash amount.">
+		effectiveRepriceDate AdjustableOrRelativeDate (1..1) <"The date to reprice the collateral">
+	output:
+		instruction PrimitiveInstruction (1..1)
+	
+	alias oldPriceQuantity:
+		tradeState -> trade -> tradableProduct -> tradeLot-> priceQuantity
+		
+	alias currentRatePrice: <"Filter interest rate price and make it into a single element">
+		oldPriceQuantity
+            extract price
+            then flatten
+            then filter priceType = PriceTypeEnum -> InterestRate
+            then only-element
+		
+	alias newPrice: <"Create the new price object.">
+        Price {
+            value: newAllinPrice,
+            unit: currentRatePrice -> unit,
+            perUnitOf: currentRatePrice -> perUnitOf,
+            priceType: currentRatePrice -> priceType,
+            priceExpression: currentRatePrice -> priceExpression,
+            composite: currentRatePrice -> composite,
+            arithmeticOperator: currentRatePrice -> arithmeticOperator,
+            cashPrice: currentRatePrice -> cashPrice,
+            datedValue: empty
+        }
+		
+	alias newPriceQuantity: <"Create the new price quantity object.">
+            PriceQuantity {
+                price: [ newPrice ],
+                quantity: empty,
+                ...
+            }
+	
+	set instruction -> split -> breakdown:
+		[ Create_TerminationInstruction( tradeState ) ]
+	
+
+
+    add instruction -> split -> breakdown: <"Sets the second part of the split to be a new contract with a new asset payout.">
+                    [
+                        PrimitiveInstruction {
+                            quantityChange: QuantityChangeInstruction {
+                                    change: [newPriceQuantity],
+                                    direction: QuantityChangeDirectionEnum -> Replace,
+                                    lotIdentifier: empty
+                                },
+                            termsChange: Create_RepriceOrAdjustmentTermsChangeInstruction(
+                                        tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+                                        effectiveRepriceDate
+                                    ),
+                            ...
+                        }
+                    ]
+
+
+		
+		
+func Create_RepriceOrAdjustmentTermsChangeInstruction: <"Creates the relevant terms change primitive instruction object for repricing a contractual product, which consists in the same terms as the original contractual product 
```

Comment:
Same comment as above. Remove and replace by generic `Create_EffectiveOrTerminationDateTermChangeInstruction `. The effective or termination date arguments can be optional (but one of them should exist, at least).

Reply:
done